File: minwook/7주차/이상한_문자_만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        String[] sList = s.split(""); // 문자열 -> 배열 변환
        int n = 0;
        for(int i=0; i<sList.length; i++){
            if(!sList[i].equals(" ")){ // 공백인지 확인
                if(n%2==0){ // 홀수면 대문자
                    answer += sList[i].toUpperCase();
                    n++; 
                }
                else{ // 짝수면 소문자
                    answer +=sList[i].toLowerCase();
                    n++;
                }
            }
            else{ // 공백이면 초기화 및 공백 더하기
                answer += sList[i];
                n=0;
            }
        }
        return answer;
    }
}

File: yongkeun/7week/이상한문자만들기.js
function solution(s) {
  let result = "";
  word = s.split(" ");
  for (let i = 0; i < word.length; i++) {
    //단어마다 돌고
    for (let j = 0; j < word[i].length; j++) {
      //단어의 알파벳마다 돌음
      if (j % 2 == 1) {
        //홀수면
        result += word[i][j].toLowerCase(); //소문자로
      } else if (j % 2 == 0) {
        //짝수면
        result += word[i][j].toUpperCase(); //대문자로
      }
    }
    if (!(i == word.length - 1)) {
      //하나의 단어를 대소문자로 변환했으면 띄어쓰기를 넣어야 하는데 마지막 단어에서는 띄어쓰기를 하지 않음
      result += " ";
    }
  }
  return result;
}
